typescript/src/shared/api.ts

```diff
@@ -43,10 +44,16 @@ class CommercetoolsAPI {
       host: apiUrl,
     };
 
+    const client = new ClientBuilder()
+      .withHttpMiddleware(httpMiddlewareOptions)
+      .withConcurrentModificationMiddleware()
+      .withUserAgentMiddleware({
+        libraryName: 'ai-gateway-mcp-essentials',
```

Comment:
This is just a **_placeholder_** and will be replaced with the actual `userAgent` string